Assets/RTLTMPro/Scripts/Runtime/LigatureFixer.cs

```diff
@@ -130,11 +145,15 @@ public static void Fix(FastStringBuilder input, FastStringBuilder output, bool f
                         }
                     } else if (isAtEnd)
                     {
-                        // Check to see if the punctuation comes at the end of the string and follows a number
+                        // Check if the punctuation comes at the end of the string and follows a number
                         bool isAfterNumber = Char32Utils.IsNumber(previousCharacter, preserveNumbers, farsi);
                         bool isSpecialPunctuation = characterAtThisIndex is '.' or '،' or '؛' or '؟';
 
-                        if (isAfterNumber && isSpecialPunctuation)
+                        // Check if the punctuation comes at the end of the string and follows a quotation mark
+                        bool isAfterQuote = Char32Utils.IsQuote(previousCharacter);
+
+                        if (isAfterNumber && isSpecialPunctuation ||
+                            isAfterQuote && isSpecialPunctuation)
```

Comment:
Double checking `isSpecialPunctuation` can be changed to `if (isSpecialPunctuation && (isAfterNumber || isAfterQuote))`